bundles/org.openhab.binding.haywardomnilogic/src/main/java/org/openhab/binding/haywardomnilogic/internal/handler/HaywardBridgeHandler.java

```diff
@@ -437,48 +437,57 @@ public synchronized String httpXmlResponse(String urlParameters) throws HaywardE
         String urlParameterslength = Integer.toString(urlParameters.length());
         String statusMessage;
 
-        try {
-            ContentResponse httpResponse = sendRequestBuilder(config.endpointUrl, HttpMethod.POST)
-                    .content(new StringContentProvider(urlParameters), "text/xml; charset=utf-8")
-                    .header(HttpHeader.CONTENT_LENGTH, urlParameterslength).send();
-
-            int status = httpResponse.getStatus();
-            String xmlResponse = httpResponse.getContentAsString();
-
-            if (status == 200) {
-                List<String> statusMessages = evaluateXPath(
-                        "/Response/Parameters//Parameter[@name='StatusMessage']/text()", xmlResponse);
-                if (!(statusMessages.isEmpty())) {
-                    statusMessage = statusMessages.get(0);
-                } else {
-                    statusMessage = httpResponse.getReason();
-                }
+        for (int retry = 0; retry <= 2; retry++) {
+            try {
+                ContentResponse httpResponse = sendRequestBuilder(config.endpointUrl, HttpMethod.POST)
+                        .content(new StringContentProvider(urlParameters), "text/xml; charset=utf-8")
+                        .header(HttpHeader.CONTENT_LENGTH, urlParameterslength).send();
+
+                int status = httpResponse.getStatus();
+                String xmlResponse = httpResponse.getContentAsString();
+
+                if (status == 200) {
+                    List<String> statusMessages = evaluateXPath(
+                            "/Response/Parameters//Parameter[@name='StatusMessage']/text()", xmlResponse);
+                    if (!(statusMessages.isEmpty())) {
+                        statusMessage = statusMessages.get(0);
+                    } else {
+                        statusMessage = httpResponse.getReason();
+                    }
 
-                if (logger.isTraceEnabled()) {
-                    logger.trace("Hayward Connection thing:  {} Hayward http command: {}", getCallingMethod(),
-                            urlParameters);
-                    logger.trace("Hayward Connection thing:  {} Hayward http response: {} {}", getCallingMethod(),
-                            statusMessage, xmlResponse);
-                }
-                return xmlResponse;
-            } else {
-                if (logger.isDebugEnabled()) {
-                    logger.debug("Hayward Connection thing:  {} Hayward http command: {}", getCallingMethod(),
-                            urlParameters);
-                    logger.debug("Hayward Connection thing:  {} Hayward http response: {} {}", getCallingMethod(),
-                            status, xmlResponse);
+                    if (logger.isTraceEnabled()) {
+                        logger.trace("Hayward Connection thing:  {} Hayward http command: {}", getCallingMethod(),
+                                urlParameters);
```

Comment:
If this is logged before sending request, it will also be logged in case an exception is thrown.

Reply:
Typically the exceptions are thrown above this line of code (httpResponse and/or evaluateXPath) such that this line would not execute.  Maybe I'm not understanding your concern.

Reply:
Sorry for the bad wording. Suggestion was to move the request logging ahead of the HTTP call, so that it would still be logged in case of an exception thrown. Actually before the loop might be even better, so that it would only be logged once in case of retry.

Reply:
Gotcha.  That would better show the program flow in the log.  The issue is, I'm also logging the server's response.

Reply:
You can still do that. :)